Applications/yOCTScanAndFindTissueSurface.m

```diff
@@ -1,44 +1,143 @@
+%       These files will be deleted after analysis is completed.
+%   dispersionQuadraticTerm - Dispersion compensation parameter.
+%   focusPositionInImageZpix - Z position pixel distance from the top where 
+%       the OCT focus is located. If not provided, the yOCTFindFocusTilledScan 
+%       function is used to determine the focus position.
+%   acceptableRange_mm - Defines the range (in millimeters) within which the 
+%       average detected tissue surface position is considered to be in focus.
+%       Used to determine the output isSurfaceInFocus. Default is 25 microns.
+%   v - Verbose mode for debugging purposes, default is false.
 % OUTPUTS:
 %   - surfacePosition_mm - 2D matrix. dimensions are (y,x). What
-%       height (mm) is image surface. Height measured from “user specified
-%       tissue interface”, higher value means deeper. See: 
+%       height (mm) is image surface. Height measured from "user specified
+%       tissue interface", higher value means deeper. See: 
 %       https://docs.google.com/document/d/1aMgy00HvxrOlTXRINk-SvcvQSMU1VzT0U60hdChUVa0/
 %   - x_mm ,y_mm are the x,y positions that corresponds to surfacePosition(y,x).
 %       Units are mm.
+%   isSurfaceInFocus - Boolean indicating whether the tissue surface is correctly
+%       positioned at the OCT focus, true if in focus, false otherwise.
 
 %% Parse inputs
 p = inputParser;
 addParameter(p,'isVisualize',false);
 addParameter(p,'xRange_mm',[-1 1]);
 addParameter(p,'yRange_mm',[-1 1]);
-addParameter(p,'resolution_mm',50e-3);
+addParameter(p,'pixel_size_um',10);
+addParameter(p,'octProbePath','probe.ini',@ischar);
+addParameter(p,'output_folder','./Surface_Analysis_Temp');
+addParameter(p,'dispersionQuadraticTerm',79430000,@isnumeric);
+addParameter(p,'focusPositionInImageZpix',NaN,@isnumeric);
+addParameter(p,'v',false);
+addParameter(p, 'acceptableRange_mm', 0.025, @isnumeric)
 
 parse(p,varargin{:});
 in = p.Results;
 xRange_mm = in.xRange_mm;
 yRange_mm = in.yRange_mm;
-resolution_mm = in.resolution_mm;
+pixel_size_um = in.pixel_size_um;
 isVisualize = in.isVisualize;
+octProbePath = in.octProbePath;
+output_folder = in.output_folder;
+dispersionQuadraticTerm = in.dispersionQuadraticTerm;
+acceptableRange_mm = in.acceptableRange_mm;
+v = in.v;
 
 %% Scan
-% Todo
+totalStartTime = datetime;  % Capture the starting time
+volumeOutputFolder = [output_folder '/OCTVolume/'];
+disp('Please adjust the OCT focus such that it is precisely at the intersection of the tissue and the coverslip.')
+fprintf('%s Scanning Volume...\n', datestr(datetime));
+scanParameters = yOCTScanTile (...
+    volumeOutputFolder, ...
+    xRange_mm, ...
+    yRange_mm, ...
+    'octProbePath', octProbePath, ...
+    'pixelSize_um', pixel_size_um, ...
+    'v',v  ...
+    );
+
+%% Check if focusPositionInImageZpix is provided, if not use yOCTFindFocusTilledScan
+if isempty(in.focusPositionInImageZpix) || isnan(in.focusPositionInImageZpix)
```

Comment:
When using Demo_ScanAndProcess_3D, default is focusPositionInImageZpix = []; I was thinking it might happen often that it's empty so it can use function to find it instead of error was what I thought, what do you think?

Reply:
Good point. I changed Demo_ScanAndProcess_3D to use NaN. it's better practice.